library/psa_crypto.c

```diff
@@ -3544,10 +3538,8 @@ psa_status_t mbedtls_psa_sign_hash_start(
     /* Ensure num_ops is zero'ed in case of context re-use. */
     operation->num_ops = 0;
 
-    /* Ensure default is set even if
-     * mbedtls_psa_interruptible_set_max_ops() has not been called. */
-    mbedtls_psa_interruptible_set_max_ops(
-        mbedtls_psa_interruptible_get_max_ops());
+    /* Ensure max_ops is set to the current value (or default). */
+    mbedtls_psa_interruptible_set_max_ops(psa_interruptible_get_max_ops());
```

Comment:
If setting `max_ops` has an effect here, we have a problem: we don't allow `mbedtls_psa_ecp_load_representation` to be interruptible.

Fortunately `mbedtls_psa_ecp_load_representation` doesn't do nontrivial calculations (for now — this will become a problem when PSA starts supporting compressed points!). So let's not worry about it. And let's not set `max_ops` here since it's misleading.

Also applies to `verify_start`.

Reply:
Actually, thanks to @mpg, I've realized that in `verify`, we are doing a slow calculation, namely `mbedtls_psa_ecp_load_public_part`, when the key is a key pair object (which only stores the private key, and has to calculate the public key before performing a public-key operation). This is a separate problem (calling `set_max_ops` in `verify_start` won't help, both because we're not using restartable functions in `mbedtls_psa_ecp_load_public_part` and because `start()` isn't supposed to return `OPERATION_INCOMPLETE`), which I've filed as https://github.com/Mbed-TLS/mbedtls/issues/7157.